test/e2e/framework.go

```diff
@@ -938,6 +944,58 @@ func (data *TestData) deployFlowAggregator(ipfixCollector string, o flowVisibili
 	if err = data.podWaitForReady(2*defaultTimeout, flowAggPod.Name, flowAggregatorNamespace); err != nil {
 		return err
 	}
+	// Check recordmetrics of Flow Aggregator to make sure Antrea-agent Pods/ClickHouse/IPFIX collector and Flow Aggregator
+	// are correctly connected
+	if err := data.getAndCheckFlowAggregatorMetrics(flowAggPod.Name); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (data *TestData) getAndCheckFlowAggregatorMetrics(podName string) error {
+	errLogs := []string{}
+	if err := wait.Poll(defaultInterval, 2*defaultTimeout, func() (bool, error) {
+		localConfig := restclient.CopyConfig(data.kubeConfig)
+		listenAddress := "localhost"
+		listenPort := apis.FlowAggregatorAPIPort
+		pf, err := portforwarder.NewPortForwarder(localConfig, flowAggregatorNamespace, podName, apis.FlowAggregatorAPIPort, "localhost", listenPort)
+		if err != nil {
+			errLogs = append(errLogs, fmt.Sprintf("error when port-forwarding Flow Aggregator Pod: %s/%s, err: %v", flowAggregatorNamespace, podName, err))
+			return false, nil
+		}
+		pf.Start()
+		defer pf.Stop()
+		localConfig.Host = net.JoinHostPort(listenAddress, fmt.Sprint(listenPort))
+		localConfig.Insecure = true
+		localConfig.CAFile = ""
+		localConfig.CAData = nil
+		clientset, err := kubernetes.NewForConfig(localConfig)
+		if err != nil {
+			errLogs = append(errLogs, fmt.Sprintf("error when creating Flow Aggregator client: %v", err))
+			return false, nil
+		}
+		jsonMetrics, err := clientset.RESTClient().
+			Get().
+			Name("recordmetrics").
+			Do(context.TODO()).Raw()
+		if err != nil {
+			errLogs = append(errLogs, fmt.Sprintf("error when requesting recordmetrics, %v", err))
+			return false, nil
+		}
+		metrics := &recordmetrics.Response{}
+		err = json.Unmarshal(jsonMetrics, metrics)
+		if err != nil {
+			errLogs = append(errLogs, fmt.Sprintf("error when decoding recordmetrics, %v", err))
+			return false, nil
+		}
+		if metrics.NumConnToCollector != defaultNumberOfNodes || !metrics.IsClickHouseRunning || !metrics.IsIPFIXRunning {
```

Comment:
you need to use the actual number of Nodes. We have this information in `clusterInfo`.

Reply:
Fixed. Thanks.